cloud/scope/powervs_cluster_test.go

```diff
@@ -3830,3 +3832,799 @@ func TestCreateVPCSubnet(t *testing.T) {
 		g.Expect(err).ToNot(BeNil())
 	})
 }
+
+func TestIsResourceCreatedByController(t *testing.T) {
+	testCases := []struct {
+		name           string
+		resourceType   infrav1beta2.ResourceType
+		clusterScope   PowerVSClusterScope
+		expectedResult bool
+	}{
+		{
+			name: "When resourceType is VPC and VPC status is nil",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
+			},
+			resourceType:   infrav1beta2.ResourceTypeVPC,
+			expectedResult: false,
+		},
+		{
+			name: "When resourceType is VPC and VPC status is not nil",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+					Status: infrav1beta2.IBMPowerVSClusterStatus{
+						VPC: &infrav1beta2.ResourceReference{
+							ControllerCreated: ptr.To(true),
+						},
+					},
+				},
+			},
+			resourceType:   infrav1beta2.ResourceTypeVPC,
+			expectedResult: true,
+		},
+		{
+			name: "When resourceType is ServiceInstance and ServiceInstance status is nil",
+			clusterScope: PowerVSClusterScope{
+				IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{},
```

Comment:
I believe this declaration is to avoid panic?

Reply:
Yes